src/java.desktop/share/classes/javax/swing/JSplitPane.java

```diff
@@ -362,6 +363,32 @@ public JSplitPane(int newOrientation,
 
     }
 
+    /**
+     * {@inheritDoc}
+     * @param orientation {@inheritDoc}
+     */
+    @Override
+    public void setComponentOrientation(ComponentOrientation orientation) {
+        super.setComponentOrientation(orientation);
+        Component leftComponent = this.getLeftComponent();
+        Component rightComponent = this.getRightComponent();
+        if (this.getComponentOrientation().equals(ComponentOrientation.RIGHT_TO_LEFT)) {
```

Comment:
Is it a good idea to use `isLeftToRight()` method to determine the component orientation? 
As per [ComponentOrientation spec](https://docs.oracle.com/en%2Fjava%2Fjavase%2F22%2Fdocs%2Fapi%2F%2F/java.desktop/java/awt/ComponentOrientation.html) 

 >Components whose view and controller code depends on orientation should use the isLeftToRight() and isHorizontal() methods to determine their behavior. They should not include switch-like code that keys off of the constants, such as:
if (orientation == LEFT_TO_RIGHT) {
   ...
 } else if (orientation == RIGHT_TO_LEFT) {
   ...
 } else {
   // Oops
 }
This is unsafe, since more constants may be added in the future and since it is not guaranteed that orientation objects will be unique.`

Reply:
good observation..will modify...

Reply:
There's a problem to it as ComponentOrientation can be LTR, RTL or UNKNOWN (ie orientation not set) so just checking for isLeftToRIght will not work as !LTR can be RTL or UNKNOWN so it will not work for the present scenario so I guess we need to use the present way only..

Reply:
>  just checking for isLeftToRIght will not work as !LTR can be RTL or UNKNOWN

Does that mean if orientation is UNKNOWN, it should be LTR by default ?

Reply:
It does seem like it...

Reply:
> There's a problem to it as ComponentOrientation can be LTR, RTL or UNKNOWN (ie orientation not set) so just checking for isLeftToRIght will not work as !LTR can be RTL or UNKNOWN so it will not work for the present scenario so I guess we need to use the present way only..

You are right as !LTR can be one of RTL or UNKNOWN but it is mentioned in [ComponentOrientation](https://github.com/kumarabhi006/jdk/blob/70f3e99016311a6520e6a7c0da4c7ff718364976/src/java.desktop/share/classes/java/awt/ComponentOrientation.java#L120) class that `isLeftToRight will return true for UNKNOWN value`.

  /**
     * Indicates that a component's orientation has not been set.
     * To preserve the behavior of existing applications,
     * isLeftToRight will return true for this value.
     */
    public static final ComponentOrientation UNKNOWN =
                    new ComponentOrientation(HORIZ_BIT|LTR_BIT|UNK_BIT);

So, I guess this should be fine to check for `!isLeftToRIght()` to add components in a RTL else in LTR.

Reply:
> It does seem like it...all textfield/textarea even if orientation not set, it writes text in LTR, right?

Mentioned in [Component](https://docs.oracle.com/en%2Fjava%2Fjavase%2F22%2Fdocs%2Fapi%2F%2F/java.desktop/java/awt/Component.html#setComponentOrientation(java.awt.ComponentOrientation)) class about UNKNOWN orientation

`At construction time, a component's orientation is set to ComponentOrientation.UNKNOWN, indicating that it has not been specified explicitly. The UNKNOWN orientation behaves the same as ComponentOrientation.LEFT_TO_RIGHT. `

Reply:
ok